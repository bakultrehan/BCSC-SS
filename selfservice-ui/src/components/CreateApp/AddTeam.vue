/* AddTeam component */
<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-btn icon @click="goBack()" :aria-label="$t('team.btnBack')">
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <v-toolbar-title>{{ $t('team.pageTitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-col cols="12">
      <TeamRoles :id="id" />
    </v-col>
    <v-col cols="12">
      <v-card flat class="ma-5">
        <v-divider></v-divider>
        <v-card-actions>
          <v-spacer></v-spacer>
          <Button @click="goBack" :aria-label="$t('team.btnBack')" secondary class="back-btn">
            {{
            $t(
            showWizardExperience()
            ? 'team.btnBack'
            : 'team.btnCancel'
            )
            }}
          </Button>
          <Button class="white--text ml-6" depressed @click="nextRedirect">
            {{
            $t(
            showWizardExperience()
            ? 'team.btnNext'
            : 'team.btnSaveChanges'
            )
            }}
          </Button>
        </v-card-actions>
      </v-card>
    </v-col>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Button from '@/Atomic/Button/Button.vue';
import TeamRoles from '@/components/TeamRoles/TeamRoles.vue';
const SharedModule = namespace('SharedModule');

@Component({
  components: {
    Button,
    TeamRoles
  }
})
export default class AddTeam extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;
  @SharedModule.Getter('isRedirectFromSummaryPage')
  public isRedirectFromSummaryPage!: boolean;

  private projectId: number = this.id || 0;

  private goBack() {
    const redirectPage = this.showWizardExperience() ? 'info' : 'summary';
    this.redirectFromSummaryPage(false);
    this.$router.push(`/project/${this.projectId}/${redirectPage}/`);
  }
  private showWizardExperience() {
    if (this.isRedirectFromSummaryPage) {
      return false;
    }
    return true;
  }

  private nextRedirect() {
    const redirectPage = this.showWizardExperience() ? 'technical' : 'summary';
    this.$router.push(`/project/${this.projectId}/${redirectPage}/`);
  }
}
</script>

<style lang="scss" scoped>
</style>
