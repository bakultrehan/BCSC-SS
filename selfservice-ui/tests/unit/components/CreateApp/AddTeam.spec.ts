import { mount, createLocalVue } from '@vue/test-utils';
import AddTeam from '@/components/CreateApp/AddTeam.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('AddTeam.vue', () => {
  let wrapper: any;
  const localVue = createLocalVue();

  localVue.use(Vuetify);
  localVue.use(Vuex);
  vuetify = new Vuetify();
  const store = new Vuex.Store({
    modules: {
      SharedModule: {
        namespaced: true,
        state: {},
        getters: {
          isRedirectFromSummaryPage: jest.fn()
        },
        actions: {
          redirectFromSummaryPage: jest.fn()
        }
      },
      TeamRolesModule: {
        namespaced: true,
        state: {},
        getters: {
          getTeamList: jest.fn(() => {
            return [];
          })
        },
        actions: {}
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(AddTeam, {
      store,
      vuetify,
      localVue,
      sync: false,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed', () => {
    wrapper = mountFunction({});
    expect(wrapper.element).toMatchSnapshot();
  });

  // it('submit form on onlick ', () => {
  //   wrapper = mountFunction({});

  //   const addTeam = jest.fn();
  //   const button = wrapper.find('.submit-team');
  //   wrapper.vm.$on('action-btn:clicked', addTeam);
  //   button.trigger('click');

  //   expect(wrapper.element).toMatchSnapshot();
  // });
});
