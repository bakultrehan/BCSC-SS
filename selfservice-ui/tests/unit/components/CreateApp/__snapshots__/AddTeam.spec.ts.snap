// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTeam.vue renders props when passed 1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-left theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="col col-12"
  >
    <div
      class="container"
    >
      <div
        class="row ma-5"
      >
        <div
          class="col col-12"
        >
          <h2
            class="text-left tab-headline"
          >
            
        
      
          </h2>
           
          <h4
            class="text-left tab-headline font-weight-medium"
          >
            
        
      
          </h4>
           
          <div
            class="v-data-table mt-5 theme--light"
          >
            <div
              class="v-data-table__wrapper"
            >
              <table>
                <thead
                  class="tbl-head"
                >
                  <tr
                    class="text-left"
                  >
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                  </tr>
                </thead>
                 
                <tbody />
              </table>
            </div>
          </div>
           
          <button
            class="btn v-btn v-btn--contained theme--light v-size--default primary normal"
            name="button"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              
            </span>
          </button>
        </div>
         
        <div
          class="col"
        >
          <div
            class="text-center"
          >
            <div
              class="v-dialog__container text-left"
              role="dialog"
            >
              <div
                class="v-dialog__content"
                role="document"
                style="z-index: 0;"
              >
                <div
                  class="v-dialog v-dialog--persistent"
                  style="width: 70%; display: none;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col col-12"
  >
    <div
      class="ma-5 v-card v-card--flat v-sheet theme--light"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions"
      >
        <div
          class="spacer"
        />
         
        <button
          class="btn back-btn v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
         
        <button
          class="btn white--text submit-team ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
          depressed=""
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddTeam.vue submit form on onlick  1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-left theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="col col-12"
  >
    <div
      class="container"
    >
      <div
        class="row ma-5"
      >
        <div
          class="col col-12"
        >
          <h2
            class="text-left tab-headline"
          >
            
        
      
          </h2>
           
          <h4
            class="text-left tab-headline font-weight-medium"
          >
            
        
      
          </h4>
           
          <div
            class="v-data-table mt-5 theme--light"
          >
            <div
              class="v-data-table__wrapper"
            >
              <table>
                <thead
                  class="tbl-head"
                >
                  <tr
                    class="text-left"
                  >
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                     
                    <th
                      class="text-left"
                    >
                      
                    </th>
                  </tr>
                </thead>
                 
                <tbody />
              </table>
            </div>
          </div>
           
          <button
            class="btn v-btn v-btn--contained theme--light v-size--default primary normal"
            name="button"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              
            </span>
          </button>
        </div>
         
        <div
          class="col"
        >
          <div
            class="text-center"
          >
            <div
              class="v-dialog__container text-left"
              role="dialog"
            >
              <div
                class="v-dialog__content"
                role="document"
                style="z-index: 0;"
              >
                <div
                  class="v-dialog v-dialog--persistent"
                  style="width: 70%; display: none;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col col-12"
  >
    <div
      class="ma-5 v-card v-card--flat v-sheet theme--light"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions"
      >
        <div
          class="spacer"
        />
         
        <button
          class="btn back-btn v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
         
        <button
          class="btn white--text submit-team ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
          depressed=""
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
